staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -475,15 +475,15 @@ func (qs *queueSet) removeTimedOutRequestsFromQueueLocked(queue *queue, fsName s
 	// now - requestWaitLimit = waitLimit
 	waitLimit := now.Add(-qs.qCfg.RequestWaitLimit)
 	for i, req := range reqs {
-		if waitLimit.After(req.arrivalTime) {
-			req.decision.SetLocked(decisionReject)
-			// get index for timed out requests
-			timeoutIdx = i
-			metrics.AddRequestsInQueues(qs.qCfg.Name, req.fsName, -1)
-			req.NoteQueued(false)
-		} else {
+		if req.ctx.Err() == nil && !waitLimit.After(req.arrivalTime) {
```

Comment:
why is htis change in here?

Reply:
while a request is waiting in p&f queue its context can time out, the timeout filter will send a timeout response to the client. but the request will still be waiting in p&f queue. I will do some more research and follow up with Mike. If there is an optimization here I can open a follow up PR.

I removed this for now.